Assets/Scripts/MineManager.cs

```diff
@@ -0,0 +1,221 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using UnityEngine.UIElements;
+using Random = UnityEngine.Random;
+
+
+public class MineManager : MonoBehaviour
+{
+    // model
+    public enum State 
+    {
+        Init,
+        Flagged,
+        Opened
+    }
+    public int  gridSize;
+    public int mineCount;
+    public int[,] gameGrid;
+    public State[,] gridStates;
+    public int opened = 0;
+    public String difficulty;
+
+    private float time = 0;
+    private float delayTime = 1f;
+    public bool endGame = false;
+    public bool win = false;
+    public bool lose = false;
```

Comment:
win / lose both imply endGame so you shouldnt store endGame separately. win and lose cannot be true at the same time so they shouldnt be stored separately. you should have a single enum to indicate what state this is in.